spec/lib/miq_automation_engine/miq_ae_object_spec.rb

```diff
@@ -74,6 +74,16 @@ def value_match(value, xml_value)
     expect(result["vms"].length).to eq(1)
   end
 
+  it "#process_args_as_attributes with mixed types and case insensitive" do
+    result = @miq_obj.process_args_as_attributes("Array::VMs" => "VmOrTemplate::#{@vm.id}",
+                                                 "Name"       => "fred")
+    expect(result["vms"]).to be_kind_of(Array)
+    expect(result["vms"].length).to eq(1)
+    expect(result["name"]).to eq("fred")
+    expect(result["Name"]).to be_nil
+    expect(result["VMs"]).to be_nil
```

Comment:
Minor, but can you move this check for `VMs` next to the `vms` tests so they are grouped together.  Makes it clearer what you are testing.